wifiphisher/common/interfaces.py

```diff
@@ -243,6 +243,36 @@ def up_ifaces(self, ifaces):
         for i in ifaces:
             card = pyw.getcard(i.get_name())
             pyw.up(card)
+    
+    def _generate_random_address(self):
```

Comment:
The method that should be part of the `NetworkAdapter` class is the `randomize_interface` (maybe rename to `randomize_mac`) that modifies an attribute of the object.

More specifically, the `NetworkAdapter` class should have two new fields: `previous_mac` and `current_mac`. During an initialization of an object these two fields should be set with the current MAC of the network card (using `macget` of PyRIC).

Now, when we randomize the mac, we should also set the `current_mac` with the newly generated mac address. During the `shutdown` of the tool, we can revert the card back to its original mac address.